### PR TITLE
Enable precompiled headers

### PR DESCRIPTION
**Pull request by woshiyifei**

### CODE REVIEW

File: exceptions.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#include <stdexcept>
+
+/// macro for convenient definition of common clang exceptions
+#define DEFINE_COMMON_CLANG_EXCEPTION(__name)                                      \
+  class __name : public std::domain_error {                                    \
+  public:                                                                      \
+    __name(const std::string &str) : std::domain_error(str) {}                 \
+  };
+
+DEFINE_COMMON_CLANG_EXCEPTION(pch_error)
```
Comment (AlexeySotkin) [inline exceptions.h:29]: In what case this exception is or should be thrown? If we don't throw it, I think we can remove this file.

File: options_compile.cpp [hunk 2]
```diff
@@ -60,10 +60,12 @@ int EffectiveOptionsFilter::s_progID = 1;
 // Options filter that validates the opencl used options
 //
 std::string EffectiveOptionsFilter::processOptions(const OpenCLArgList &args,
+                                                   const OpenCLArgList &argsEx,
                                                    const char *pszOptionsEx,
                                                    ArgsVector &effectiveArgs) {
   // Reset args
   int iCLStdSet = 0;
+  int fp64Enable = 0;
```
Comment (AlexeySotkin) [inline options_compile.cpp:68]:
```suggestion
  boole isFp64Enable = false;
```

File: options_compile.cpp [hunk 3]
```diff
@@ -192,6 +194,49 @@ std::string EffectiveOptionsFilter::processOptions(const OpenCLArgList &args,
 
   effectiveArgs.push_back(szTriple);
 
+  for (OpenCLArgList::const_iterator it = argsEx.begin(), ie = argsEx.end();
```
Comment (AlexeySotkin) [inline options_compile.cpp:197]: Can we use 
```
fp64Enable = argsEx.hasArgNoClaim(OPT_COMPILE_cl_khr_fp64);
```
instead?

File: 0001-Enable-procompiled-headers.patch [hunk 4]
```diff
@@ -0,0 +1,448 @@
+From e3915a9be0e4d90476264ab8245a3ac02a6d4b4b Mon Sep 17 00:00:00 2001
```
Comment (AlexeySotkin) [inline 0001-Enable-procompiled-headers.patch:1]: I don't understand why do we need this patch-file here.

File: exceptions.h (hunk 1 above)
Comment (woshiyifei) [reply exceptions.h]: Removed

File: 0001-Enable-procompiled-headers.patch (hunk 4 above)
Comment (woshiyifei) [reply 0001-Enable-procompiled-headers.patch]: Removed

File: options_compile.cpp (hunk 3 above)
Comment (woshiyifei) [reply options_compile.cpp:197]: According to my local test, cl-ext=...,... info is contained in pszOptionsEx, so I added check for args in pszOptionsEx.

Comment (woshiyifei) [reply options_compile.cpp]: I've updated the patch as request.

**Pull request merged**